content/tutorial/tutorial.md

```diff
@@ -1042,20 +1049,21 @@ Let's `map` over the `history` in the Game's `render` method:
   }
 ```
 
-**[View the full code at this point](https://codepen.io/gaearon/pen/EmmGEa?editors=0010)**
+**[अभी तक का पूरा कोड देखें](https://codepen.io/gaearon/pen/EmmGEa?editors=0010)**
 
-For each move in the tic-tac-toes's game's history, we create a list item `<li>` which contains a button `<button>`. The button has a `onClick` handler which calls a method called `this.jumpTo()`. We haven't implemented the `jumpTo()` method yet. For now, we should see a list of the moves that have occurred in the game and a warning in the developer tools console that says:
+टिक-टैक्-टो के खेल के इतिहास में प्रत्येक चाल के लिए, हम एक सूची आइटम `<li>` बनाते हैं जिसमें एक बटन होता है `<button>`। बटन में एक  `onClick` हैंडलर होता है, जो कि `this.jumpTo()` नामक एक मेथड कॉल करता है। 
+हमने अभी तक `jumpTo()` मेथड बनाई नहीं की है। अभी के लिए, हमें खेल में होने वाली चालों की सूची दिखनी चाहिए और डेवलपर टूल्स कंसोल में एक चेतावनी दी गई है जो कहती है:
 
->  Warning:
->  Each child in an array or iterator should have a unique "key" prop. Check the render method of "Game".
+>  चेतावनी:
+>  एक array या पुनरावृत्त में प्रत्येक चाइल्ड के पास एक अद्वितीय "कुंजी" प्रोप होना चाहिए। "Game" की रेंडर मेथड की जाँच करें।
 
-Let's discuss what the above warning means.
+आइए चर्चा करें कि उपरोक्त चेतावनी का क्या मतलब है।
 
-### Picking a Key {#picking-a-key}
+### एक कुंजी उठाना {#picking-a-key}
 
-When we render a list, React stores some information about each rendered list item. When we update a list, React needs to determine what has changed. We could have added, removed, re-arranged, or updated the list's items.
+जब हम एक सूची सौंपते हैं, तो React प्रत्येक रेंडर सूची आइटम के बारे में कुछ जानकारी संग्रहीत करती है। जब हम किसी सूची को अपडेट करते हैं, तो React को यह निर्धारित करने की आवश्यकता होती है कि क्या बदल गया है। हम सूची के आइटम को जोड़, हटा, पुन: व्यवस्थित या अपडेट कर सकते थे।
```

Comment:
render not being translated

Reply:
revisit paragraph

Reply:
```
जब हम एक लिस्ट render करते हैं, React हर rendered लिस्ट आइटम की कुछ इनफार्मेशन स्टोर करता है। जब हम एक लिस्ट को अपडेट करते हैं, तो React को ये पता करना होता है की क्या अपडेट हुआ है। हमनें लिस्ट के आइटम्स को ऐड, रिमूव, री-अरेैंज या अपडेट किया हो सकता है।
```

how about this?

Reply:
हमनें लिस्ट के आइटम्स को जोड़, हटा, पुन: व्यवस्थित या अपडेट किया हो सकता है। could be improved upon I think, the rest is good

Reply:
> हमनें लिस्ट के आइटम्स को जोड़, हटा, पुन: व्यवस्थित या अपडेट किया हो सकता है। could be improved upon I think, the rest is good

`हमनें लिस्ट में कुछ भी बदलाव किये हो सकते हैं` ?

Reply:
The meaning of the line "We could have added, removed, re-arranged, or updated the list's items." vs that would be a little different, no?

Reply:
I can't think of any better translation. Changing `removed`, `re-arranged`, `updated` hindi translations are becoming complex. Any suggestions?

Reply:
leave rearrange and update as is in hindi?